Automattic/wp-calypso
# Build: Dynamically import Flag SVGs

See https://github.com/Automattic/wp-calypso/issues/26296#issuecomment-407878702

> I've been talking to @blowery, and he brought up using an SVG loader for webpack plus[ dynamic imports](https://webpack.js.org/guides/code-splitting/#dynamic-imports), which sounds like an elegant solution.

TODO:
- [x] Make work
- [x] Make dynamic
- [x] `svg-inline-loader` probably needs to be a dep, not a devDep
- [x] Drop flag copying from `webpack.config` 87bb42d#diff-11e9f7f953edc64ba14b0cc350ae7b9d

**Notes**
We're using the inline loader syntax here. That begs the question: why not use a new rule in the wepback config? It's possible that such would be a good path forward but I'm not entirely sure. Only a single file is using SVG imports and this has been true for a very long time in the project's history. Further, since there are different ways to import SVG I felt that the practice of importing as a data URI would be more esoteric and surprising than I think people would expect. Therefore I think there's reasonable warrant to leave this inline - open to thoughts. -- @dmsnell 

[ICFY](http://iscalypsofastyet.com/branch?branch=update/fun-with-flags) looks bad, but keep in mind that we're getting rid of `public/images/flags/` instead, which had the same SVGs. The loader adds a couple of bytes to each SVG, but they're still loaded dynamically, i.e. we aren't inflating other bundles.

To test:
* `npm run distclean && npm i && npm start`
* Check phone input dropdown ('Form Media Phone Input') at http://calypso.localhost:3000/devdocs/design/form-fields
* Verify that after ending Calypso, no flag assets are in `flags/images/*`. Try a wpcalypso build in Docker to be sure. (#26141 thus being resolved)
